### PR TITLE
lua*Packages improvements

### PR DESCRIPTION
**Pull request by vcunat**
###### Motivation for this change

I put everything inside a single PR, because there are multiple dependencies among the commits.  Feel free to review just parts.  I can split this later, in case some part is contentious.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] N/A Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of **some** pkgs that depend on this change ~~using `nix-shell -p nix-review --run "nix-review wip"`~~
- [ ] N/A Tested execution of all binary files (usually in `./result/bin/`)
- [ ] _Determined the impact on package closure size (by running `nix path-info -S` before and after)_
     Some packages newly have runtime dependency on lua interpreter or other lua packages it depends on, but that's probably not considered a regression.  Also, the luarocks builder additionally includes some docs, but that's not big.
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md), I think.

---

### CODE REVIEW

**Comment by vcunat** [comment]
@teto might be interested.

File: pkgs/servers/dns/knot-resolver/default.nix [hunk 1]
```diff
@@ -75,26 +75,24 @@ unwrapped = stdenv.mkDerivation rec {
...
     allowSubstitutes = false;
   }
   ''
-    mkdir -p "$out/sbin" "$out/share"
-    makeWrapper '${unwrapped}/sbin/kresd' "$out"/sbin/kresd \
-      --set LUA_PATH  '${concatStringsSep ";" (map getLuaPath  luaPkgs)}' \
-      --set LUA_CPATH '${concatStringsSep ";" (map getLuaCPath luaPkgs)}'
+    mkdir -p "$out"/{bin,share}
+    makeWrapper '${unwrapped}/bin/kresd' "$out"/bin/kresd \
+      --set LUA_PATH  "$LUA_PATH" \
```
Comment (teto) [inline pkgs/servers/dns/knot-resolver/default.nix]: can't you remove this instead ?

**Comment by teto** [comment]
@GrahamcOfBorg build luaPackages.luacyrussasl luaPackages.binaryheap luaPackages.http

File: pkgs/servers/dns/knot-resolver/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/servers/dns/knot-resolver/default.nix]: Not by itself.  Someone has to set the variables in the wrapper.  Maybe I misunderstand you?

**Comment by teto** [comment]
Looks fine to me. seems like luacyrussasl doesn't like darwin but hopefully luarocks-nix will translate the supported formats to nix soon so no need to delay this further. feel free to merge.

File: pkgs/servers/dns/knot-resolver/default.nix (hunk 1 above)
Comment (teto) [reply pkgs/servers/dns/knot-resolver/default.nix]: right.

**Pull request merged**